pyccel/ast/cwrapper.py

```diff
@@ -528,12 +548,56 @@ def pybool_to_bool(cast_function_name):
                        body      = cast_function_body,
                        results   = [cast_function_result])
 
+def pyarray_to_ndarray(cast_function_name):
+    cast_function_argument = Variable(dtype=PyccelPyArrayObject(), name='o',
+                                is_pointer=True)
+    cast_function_result   = Variable(dtype=NativeInteger(), name = 'c',
+                                rank=1)
+
+    nd          = DottedVariable(NativeInteger(), 'nd',
+                                lhs=cast_function_result.name)
+    raw_data    = DottedVariable(NativeGeneric(), 'raw_data',
+                                lhs=cast_function_result.name, rank=1)
+    shape       = DottedVariable(NativeInteger(), 'shape',
+                                lhs=cast_function_result.name, is_pointer=True)
+    type_size   = DottedVariable(NativeInteger(), 'type_size',
+                                lhs=cast_function_result.name)
+    strides     = DottedVariable(NativeInteger(), 'strides',
+                                lhs=cast_function_result.name)
+    arr_type    = DottedVariable(NativeInteger(), 'type',
+                                lhs=cast_function_result.name)
+    length      = DottedVariable(NativeInteger(), 'length',
+                                lhs=cast_function_result.name)
+    buffer_size = DottedVariable(NativeInteger(), 'buffer_size',
+                                lhs=cast_function_result.name)
+    is_view     = DottedVariable(NativeInteger(), 'is_view',
+                                lhs=cast_function_result.name)
```

Comment:
Could you make this block a bit more readable please? Moreover, you are calling the same functions many times.

You may want to copy and paste this:
```python
    Int = NativeInteger()
    Gen = NativeGeneric()
    lhs = cast_function_result.name

    nd          = DottedVariable(Int,          'nd', lhs=lhs)
    raw_data    = DottedVariable(Gen,    'raw_data', lhs=lhs, rank=1)
    shape       = DottedVariable(Int,       'shape', lhs=lhs, is_pointer=True)
    type_size   = DottedVariable(Int,   'type_size', lhs=lhs)
    strides     = DottedVariable(Int,     'strides', lhs=lhs)
    arr_type    = DottedVariable(Int,    'arr_type', lhs=lhs)
    length      = DottedVariable(Int,      'length', lhs=lhs)
    buffer_size = DottedVariable(Int, 'buffer_size', lhs=lhs)
    is_view     = DottedVariable(Int,     'is_view', lhs=lhs)

```